src/odemis/driver/simstreakcam.py

```diff
@@ -0,0 +1,544 @@
+        self._exp_time = 0.1  # 100 msec
+        self.exposureTime = model.FloatContinuous(self._exp_time, range_exp, unit="s", setter=self._setCamExpTime)
+        self._metadata[model.MD_EXP_TIME] = self.exposureTime.value
+
+        self.readoutRate = model.VigilantAttribute(425000000, unit="Hz", readonly=True)  # MHz
+        self._metadata[model.MD_READOUT_TIME] = 1 / self.readoutRate.value  # s
+
+        # spectrograph VAs after readout camera VAs
+        if self._spectrograph:
+            logging.debug("Starting streak camera with spectrograph.")
+            self._spectrograph.position.subscribe(self._updateWavelengthList, init=True)
+
+        # for synchronized acquisition
+        self._sync_event = None
+        self.softwareTrigger = model.Event()
+
+        # Simple implementation of the flow: we keep generating images and if
+        # there are subscribers, they'll receive it.
+        self.data = SimpleStreakCameraDataFlow(self._start, self._stop, self._sync)
+        self._generator = None
+
+    def _updateWavelengthList(self, _=None):
+        npixels = self.resolution.value[0]  # number of pixels, horizontal is wavelength
+        # pixelsize VA is sensor px size without binning and magnification
+        pxs = self.pixelSize.value[0] * self.binning.value[0] * self._metadata.get(model.MD_LENS_MAG, 1.0)
+        wll = self._spectrograph.getPixelToWavelength(npixels, pxs)
+        self._metadata[model.MD_WL_LIST] = wll
+
+    def _setBinning(self, value):
+        """
+        value (2-tuple int)
+        Called when "binning" VA is modified. It actually modifies the camera binning.
+        """
+        prev_binning, self._binning = self._binning, value
+
+        # adapt resolution
+        # TODO check if really necessary: why not just call resolutionFitter
+        # TODO self._binning is already updated and shape does not change
+        change = (prev_binning[0] / self._binning[0],
+                  prev_binning[1] / self._binning[1])
+        old_resolution = self.resolution.value
+        new_res = (int(round(old_resolution[0] * change[0])),
+                   int(round(old_resolution[1] * change[1])))
+
+        # fit
+        self.resolution.value = new_res
+
+        self._metadata[model.MD_BINNING] = self._binning  # update MD
+
+        if self._spectrograph:
+            self._updateWavelengthList()  # update WavelengthList when changing binning
+
+        return self._binning
+
+    def _setResolution(self, _=None):
+        """Sets the resolution VA.
+        So far the full field of view is always used. Therefore, resolution only changes with binning
+        or magnification."""
+
+        # Note: we can keep it simple as long as we do not provide to change the sensor size yet...
+        resolution = self._shape[:2]
+        new_res = (int(resolution[0] // self._binning[0]),
+                    int(resolution[1] // self._binning[1]))  # floor division: not below zero
```

Comment:
a floor division doesn't mean it doesn't go below zero, it means that it rounds down to the closest integer below the value.
Ex: 50//40 -> 1.25 -> 1
50//60 -> 0.833 -> 0